### PR TITLE
fby35: gl: support platform config initialization

### PR DESCRIPTION
**Pull request by BonnieLo**
Summary:
- Support Great Lakes platform configuartion initialization
  The platform configuration is read from SB CPLD.
  - Serverboard revision
  - 1OU status
  - 2OU status
  The HSC module type can be judged through BIC GPIOA0 "HCS_TYPE."
- Support sensor table v20230203

Dependency: #884  #880 

Test Plan:
Build code: pass
Check sensor reading: pass
Check 2OU present status with 2OU expansion board: pass
Check 2OU present status without 2OU expansion board: pass

### CODE REVIEW

File: meta-facebook/yv35-gl/src/platform/plat_class.c [hunk 1]
```diff
@@ -40,7 +46,232 @@ CARD_STATUS get_1ou_status()
...
+				if ((cnt == ARRAY_SIZE(_1ou_card_mapping_table)) &&
+				    (_1ou_status.card_type == TYPE_1OU_UNKNOWN)) {
+					LOG_ERR("unknown the 1OU card type, the voltage of ADC channel-6 is %fV",
+					       voltage);
+				}
+			}
+		}
+	}
+
+	/* TODO:
```
Comment (garnermic) [inline meta-facebook/yv35-gl/src/platform/plat_class.c:263]: Remove todo comment.

File: meta-facebook/yv35-gl/src/platform/plat_class.c [hunk 2]
```diff
@@ -40,7 +46,232 @@ CARD_STATUS get_1ou_status()
...
+					msg.data[1] = _1ou_status.card_type;
+					status = i2c_master_write(&msg, retry);
+					if (status)
+						LOG_ERR("failed to set 1ou card type, ret %d", status);
+					break;
+				}
+				if ((cnt == ARRAY_SIZE(_1ou_card_mapping_table)) &&
+				    (_1ou_status.card_type == TYPE_1OU_UNKNOWN)) {
+					LOG_ERR("unknown the 1OU card type, the voltage of ADC channel-6 is %fV",
+					       voltage);
```
Comment (garnermic) [inline meta-facebook/yv35-gl/src/platform/plat_class.c:257]: Should there be a return here?

File: meta-facebook/yv35-gl/src/platform/plat_isr.c [hunk 3]
```diff
@@ -13,16 +13,493 @@
...
+		return;
+	}
+	if (k_work_cancel_delayable(&SLP3_work) != 0) {
+		LOG_ERR("[%s] Failed to cancel delayable work", __func__);
+	}
+}
+
+K_WORK_DELAYABLE_DEFINE(set_DC_on_5s_work, set_DC_on_delayed_status);
+K_WORK_DELAYABLE_DEFINE(set_DC_off_10s_work, set_DC_off_delayed_status);
+/*
```
Comment (garnermic) [inline meta-facebook/yv35-gl/src/platform/plat_isr.c:96]: Remove todo

File: meta-facebook/yv35-gl/src/platform/plat_isr.c [hunk 4]
```diff
@@ -13,16 +13,493 @@
...
+
+		if (k_work_cancel_delayable(&set_DC_off_10s_work) != 0) {
+			LOG_ERR("Cancel set dc off delay work fail");
+		}
+		set_DC_off_delayed_status();
+	} else {
+		set_DC_on_delayed_status();
+		k_work_schedule(&set_DC_off_10s_work, K_SECONDS(DC_OFF_10_SECOND));
+
+		/*
```
Comment (garnermic) [inline meta-facebook/yv35-gl/src/platform/plat_isr.c:116]: Remove todo

**Review by garnermic — CHANGES_REQUESTED** [review]

File: meta-facebook/yv35-gl/src/platform/plat_class.c (hunk 1 above)
Comment (RenChen-wiwynn) [reply meta-facebook/yv35-gl/src/platform/plat_class.c]: Removed. 
Please review it again, thanks.

File: meta-facebook/yv35-gl/src/platform/plat_isr.c (hunk 3 above)
Comment (RenChen-wiwynn) [reply meta-facebook/yv35-gl/src/platform/plat_isr.c]: Removed.
Please review it again, thanks.

File: meta-facebook/yv35-gl/src/platform/plat_isr.c (hunk 4 above)
Comment (RenChen-wiwynn) [reply meta-facebook/yv35-gl/src/platform/plat_isr.c]: Removed.
Please review it again, thanks.

File: meta-facebook/yv35-gl/src/platform/plat_class.c [hunk 5]
```diff
@@ -40,7 +46,232 @@ CARD_STATUS get_1ou_status()
...
+	{ 1.0, RANGE, TYPE_1OU_VERNAL_FALLS_WITH_TI },
+	{ 1.26, RANGE, TYPE_1OU_WAIMANO_FALLS },
+	{ 1.5, RANGE, TYPE_1OU_EXP_WITH_NIC },
+	{ 1.75, RANGE, TYPE_1OU_VERNAL_FALLS_WITH_AST },
+	{ 2.0, RANGE, TYPE_1OU_KAHUNA_FALLS },
+};
+
+bool get_adc_voltage(int channel, float *voltage)
+{
+	if (!voltage) {
```
[the remark this answers is not in the document]
Comment (RenChen-wiwynn) [reply meta-facebook/yv35-gl/src/platform/plat_class.c]: Thanks for your comment.
Already fixed. Please review it, thanks.

File: common/service/ipmi/oem_1s_handler.c [hunk 6]
```diff
@@ -1105,6 +1106,44 @@ __weak void OEM_1S_GET_SET_GPIO(ipmi_msg *msg)
...
+		}
+		else {
+			msg->completion_code = CC_UNSPECIFIED_ERROR;
+			msg->data_len = 0;
+		}
+		break;
+	case SET_VGPIO_STATUS:
+		if (msg->data_len != 3)
+			msg->completion_code = CC_INVALID_LENGTH;
+
```
[the remark this answers is not in the document]
Comment (RenChen-wiwynn) [reply common/service/ipmi/oem_1s_handler.c:1136]: Thanks for your comment.
Already fixed. Please review it, thanks.

File: common/service/ipmi/oem_1s_handler.c [hunk 7]
```diff
@@ -1105,6 +1106,44 @@ __weak void OEM_1S_GET_SET_GPIO(ipmi_msg *msg)
...
 
+__weak void OEM_1S_GET_SET_BIC_VGPIO(ipmi_msg *msg)
+{
+	CHECK_NULL_ARG(msg);
+
+	switch (msg->data[0]) {
+	case GET_VGPIO_DIRECTION_AND_STATUS:
+		if (msg->data_len != 2)
+			msg->completion_code = CC_INVALID_LENGTH;
+
```
[the remark this answers is not in the document]
Comment (RenChen-wiwynn) [reply common/service/ipmi/oem_1s_handler.c:1120]: Thanks for your comment.
Already fixed. Please review it, thanks.

File: meta-facebook/yv35-gl/src/platform/plat_class.c [hunk 8]
```diff
@@ -40,7 +46,232 @@ CARD_STATUS get_1ou_status()
...
+	 */
+	msg.bus = SB_CPLD_BUS;
+	msg.target_addr = SB_CPLD_ADDR;
+	msg.tx_len = 1;
+	msg.rx_len = 1;
+	msg.data[0] = SB_CPLD_REG_BOARD_REVISION_ID;
+	status = i2c_master_read(&msg, retry);
+	if (status)
+		LOG_ERR("failed to get board revision from cpld, ret %d", status);
+	else
```
[the remark this answers is not in the document]
Comment (garnermic) [reply meta-facebook/yv35-gl/src/platform/plat_class.c:169]: This comment hasn't been resolved yet.

File: meta-facebook/yv35-gl/src/platform/plat_class.c [hunk 9]
```diff
@@ -40,7 +46,232 @@ CARD_STATUS get_1ou_status()
...
+					    (voltage <
+					     typical_voltage + (typical_voltage * 0.05))) {
+						_1ou_status.card_type =
+							_1ou_card_mapping_table[cnt].card_type;
+					}
+					break;
+				default:
+					LOG_ERR("unknown condition 0x%x",
+					       _1ou_card_mapping_table[cnt].condition);
+					break;
```
[the remark this answers is not in the document]
Comment (garnermic) [reply meta-facebook/yv35-gl/src/platform/plat_class.c:238]: This comment hasn't been resolved yet.

**Review by garnermic — CHANGES_REQUESTED** [review]

**Pull request closed without merging**